accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,58 @@
+# Simplify .NET Output Paths
+
+We'd like to simplify the output paths for .NET 8.
+
+Currently, the default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+This is rather complicated an inconsistent, which can make for a poor first impression of the .NET platform.
+
+We'd like to try to improve the output path structure.  Desired qualities include:
+
+- Simple and consistent
+- Avoid cases where one output path is nested inside of another one
+- Avoid excessively deep directory heirarchies
+- Get rid of `obj` folder in project root
+
+## Proposed behavior
+
+Projects targeting .NET and higher will by default use a new output path format.  The output path will consist of the following 3 nested folders
+
+- `bin` - All output (including intermediate output, will go under this folder)
+- Output Type - Such as `build`, `publish`, `obj`, or `packages`
+- Pivots - This will at minimum be the `Configuration`, such as `Debug` or `Release`. Other pivots such as `TargetFramework` or `RuntimeIdentifier` may also be included, and the pivots will be joined by the underscore (`_`) character
```

Comment:
It would be awesome if we could specify making all of these pivots lowercase, `cd`'ing into the directory on the command line is annoying since I always need to remember to uppercase `Debug`

Reply:
I'd only lowercase when building the output path string, not everywhere inside of msbuild.

Reply:
I don't believe Configurations in MSBuild are case-sensitive, since MSBuild comparisons are case-insensitive.  They are case-preserving like all MSBuild properties are by default, and end up being partly case-sensitive if the file system is case-sensitive.

We had [some discussion](https://github.com/dotnet/sdk/issues/375) of this for .NET Core 1.0, it looks like we thought it would be a good idea to lowercase the configurations when constructing the paths, but we didn't end up implementing it in time.